### ISSUE TITLE
Does every component need to be wrapped with view()?

### ISSUE THREAD

**Issue by mariushab** [RisingStack/react-easy-state#161]
Hi, I just came along this library and it looks very interesting.

However, for me the 
`Wrap ALL of your components with view - including class and function ones - even if they don't seem to directly use a store.`
part of the documentation is a little bit confusing.

Consider the following example:

````javascript
const user = store({ name: 'Bob' });

const ParenComponent = view(() => (
    <>
        <p>{user.name}</p>
        <ChildComponent />
    </>
));

const ChildComponent = () => (
    <p>This would probably be a fancy component which uses a lot of styles and/or is reused a lot</p>
);
````

If I follow the documentation, I would also have to wrap the ChildComponent with the `view()` HOC. I can not imagine that this would be the case, and if I am wrong, it would be nice to know why.

Maybe you could add a example like this to the documentation, and explicitly state wether the ChildComponent needs to be wrapped with `view()` or not.
At least that would clarify the situation for me, and I think there may be other users who have the same question.

**Comment by solkimicreb** [comment]
You are right about this. The component in your example does not need to be wrapped with `view`.

The "wrap ALL components" header is a beginner-friendly safeguard. There used to be quite a lot of issues about components not re-rendering because of missing `view` wrappers in the past.

I kept the original header text but added some more context to the dropdown section. Could you check if it is informative enough now?

Another question: Is it immediately clear in the docs that the headers - like "wrap ALL your components ..." are expandable to show more context?

**Comment by bguerout** [comment]
Hello,

I was also a bit surprised by the sentence `Wrap ALL of your components with view - including class and function ones - even if they don't seem to directly use a store.`

What about : `Every component and its parents must be wrapped with view` 

Note that I found very comfortable to have all the documentation on a single page.
However, I didn't realize right away that we could unfold the headers.

**Comment by solkimicreb** [comment]
> May be you can replace it by : Every component that is using a store or part of a store inside its render must be wrapped with view or may add a example about and provide a example to explain why sometimes components don't seem to directly use a store (but need to be wrapped).

This example is currently inside the expanded section (if you click on the header). I am still not sure about having the header sentence though 😄 It is still a best practice to wrap all components (as the expanded state explains) and I worded it like that for a good reason (to be beginner friendly).

**Comment by mariushab** [comment]
Yes, the added example makes it more clear, thank you. 

Also the explanation why it is recommended to wrap every component with `view()` helps me to better understand the principles.

Regarding the expendable headers, I only unfolded them when I read the documentation for the second or third time. I assumed that they were expendable all the time, but the first time I was to "lazy" to click on them and just wanted to read the main sections. Hope that helps you.
